src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -138,7 +141,7 @@
         required    []]
     (first (cli! cli-args specs required))))
 
-(defn run-lifecycle-fn!
+(schema/defn ^:always-validate run-lifecycle-fn! :- a/TrapperkeeperAppContext
```

Comment:
I wasn't sure if we have anything calling this which expects it to return anything - as opposed to the input `app-context` just being updated as a side effect of calling this function.  Are you expecting that the return value from this function is something that a caller should formally expect to receive back?  If not, would it be better to just not document this - so that the implementation is theoretically "freer" to be changed to do something else later if we want to?